### PR TITLE
Refactor evalOnDemand

### PR DESCRIPTION
**Pull request by vmcj**
This should make the code both easier to understand and even removes a minor bug.

No functional change intended (as we would never hit that bug in practice)

### CODE REVIEW

File: webapp/src/Entity/ContestProblem.php [hunk 1]
```diff
@@ -75,7 +76,7 @@ class ContestProblem
         options: ['comment' => 'Whether to do lazy evaluation for this problem; if set this overrides the global configuration setting', 'unsigned' => true]
     )]
     #[Serializer\Exclude]
-    private ?int $lazyEvalResults = null;
+    private int $lazyEvalResults = 0;
```
Comment (nickygerritsen) [inline webapp/src/Entity/ContestProblem.php:79]: We need a migration that:

* Updates existing `null` entries to `0`
* Makes the field non nullable

File: webapp/src/Controller/API/ProblemController.php [hunk 2]
```diff
@@ -332,8 +332,8 @@ public function linkProblemAction(
 
         $contest = $this->em->getRepository(Contest::class)->find($this->getContestId($request));
 
-        $lazyEvalResults = null;
-        if ($contestProblemPut->lazyEvalResults !== null) {
+        $lazyEvalResults = 0;
+        if ($contestProblemPut->lazyEvalResults !== DOMJudgeService::LAZY_DEFAULT) {
             $lazyEvalResults = (int)$contestProblemPut->lazyEvalResults;
         }
```
Comment (nickygerritsen) [inline webapp/src/Controller/API/ProblemController.php:338]: Why not:
```suggestion
$lazyEvalResults = ($contestProblemPut->lazyEvalResults;
```

It's always an int now with a default of 0.

File: webapp/src/Entity/ContestProblem.php [hunk 3]
```diff
@@ -193,17 +194,27 @@ public function getApiColor(): ?string
...
+    public function getLazyEvalResults(): int
     {
         return $this->lazyEvalResults;
     }
 
+    public function determineOnDemand(int $config_lazy_eval_results) {
+        if ($this->lazyEvalResults === DJS::EVAL_DEMAND) {
+            return true;
+        }
+        if ($this->lazyEvalResults === DJS::EVAL_DEFAULT && $config_lazy_eval_results) {
```
Comment (nickygerritsen) [inline webapp/src/Entity/ContestProblem.php:212]: Shouldn't this be
```suggestion
        if ($this->lazyEvalResults === DJS::EVAL_DEFAULT && $config_lazy_eval_results === DJS:: EVAL_DEMAND) {
```
?

Comment (vmcj) [reply webapp/src/Entity/ContestProblem.php]: Yes

File: webapp/src/Controller/API/ProblemController.php (hunk 2 above)
Comment (vmcj) [reply webapp/src/Controller/API/ProblemController.php:338]: If only I actually committed what I discussed.

File: webapp/src/Entity/ContestProblem.php (hunk 1 above)
Comment (vmcj) [reply webapp/src/Entity/ContestProblem.php:79]: Done.

**Review by nickygerritsen — APPROVED** [review]

**Pull request merged**